roboflow/core/workspace.py

```diff
@@ -74,6 +75,34 @@ def project(self, project_name):
 
         return Project(self.__api_key, dataset_info, self.model_format)
 
+    def clip_compare(self, dir="", image_ext=".png", target_image=""):
+        """
+        @params:
+            dir: (str) = name reference to a directory of images for comparison
+            image_ext: (str) = file format for expected images (don't include the . before the file type name)
+            target_image: (str) = name reference for target image to compare individual images from directory against
+
+            returns: (dict) = a key:value mapping of image_name:comparison_score_to_target
+        """
+
+        assert isinstance(dir, str), "valid directory string required!"
+        assert isinstance(image_ext, str), "valid image ext string required!"
+        assert isinstance(target_image, str), "valid target image string required!"
+
+        # list to store comparison results in
+        comparisons = []
+
+        # grab all images in a given directory with ext type
+        for image in glob.glob(f"./{dir}/*{image_ext}"):
+            # compare image
+            similarity = clip_encode(image, target_image)
+
+            # map image name to similarity score
+            comparisons.append({image: similarity})
+
+        # TODO - sort all comparisons to find best matches?
```

Comment:
```suggestion
comparisons = sorted(comparisons, key=lambda item: -list(item.values())[0])
```
The data representation is weird, you should have something like

```
[ 
   { "name" : str
    "similarity: float}, 
       { "name" : str
    "similarity: float},
   ...
]

Not sure why you decided to have dynamic key names (depending on image name)

Reply:
Nairwita (developer on the Walmart team) and I white boarded the original resulting obj to be queried by filename so you could take random samples from a range easier--hence the dynamic key names for O(1) lookup.

Since you put in the sorting command I think we'd still be able to accomplish this with your recommended code format and use the resulting list indexes as a way to query instead. We don't really care about how we look things up so long as we can sample things like `top 10 matches`, `10 matches randomly sampled from the 70th to 90th percentile of comparisons`, etc.